lib/galaxy/webapps/galaxy/api/tool_shed_repositories.py

```diff
@@ -378,20 +448,20 @@ def repair_repository_revision( self, trans, payload, **kwd ):
 
     def __parse_repository_from_payload( self, payload, include_changeset=False ):
         # Get the information about the repository to be installed from the payload.
-        tool_shed_url = payload.get( 'tool_shed_url', '' )
-        if not tool_shed_url:
+        tool_shed_url = payload.get( 'tool_shed_url', None )
+        name = payload.get( 'name', None )
+        owner = payload.get( 'owner', None )
+        if tool_shed_url is None:
             raise exceptions.RequestParameterMissingException( "Missing required parameter 'tool_shed_url'." )
-        name = payload.get( 'name', '' )
-        if not name:
+        if name is None:
             raise exceptions.RequestParameterMissingException( "Missing required parameter 'name'." )
-        owner = payload.get( 'owner', '' )
-        if not owner:
+        if owner is None:
```

Comment:
are empty names permitted? If not, then we should probably retain the original code here or throw a different exception